### ISSUE TITLE
Falco 0.36.1 and 0.36.0 segfault sometimes with syscall (evt null)

### ISSUE THREAD

**Issue by max-frank** [falcosecurity/falco#2878]
**Describe the bug**

After the upgrade to 0.36.0 (and consecutively 0.36.1) we noticed that our falco daemonsets were restarting multiple times overnight. Note that this is in a low traffic cluster only used for testing the falco deployment so should be unrelated to K8S metadata performance issues.

Pods seem to exit with CODE: 139 seemingly randomly.  To investigate we built falco with debug release in the orignal pod environment and let it run until it crashed. See the stack trace and logs from one such below.

As you can see in the stack trace the cause of the segfault seems to be that the `evt` pointer seems to be null during the [get_source_idx call](https://github.com/falcosecurity/falco/blame/master/userspace/falco/app/actions/process_events.cpp#L280) in the main inspect loop.  Looking at the code the only reason this could happen is if the [sinsp::next](https://github.com/falcosecurity/libs/blame/5b4e28d44f9c87aa23f24c53c6bfb696b2b47635/userspace/libsinsp/sinsp.cpp#L1221) at the beginning of the loop set the pointer to null. If this is not expected behaviour then this might be a bug in the libs rather then a bug in falco caused by a missing null check.

**How to reproduce it**

Start falco with syscall source and wait. The pod will eventually crash.

**Expected behaviour**

No crashes happen.

**Environment**

GKE 1.26.6-gke.1700 (COS)

- Falco version:
```
{"default_driver_version":"6.0.1+driver","driver_api_version":"5.0.0","driver_schema_version":"2.0.0","engine_version":"26","falco_version":"0.36.1","libs_version":"0.13.2","plugin_api_version":"3.1.0"}
```
Falco version (debug build):
```
{"default_driver_version":"6.0.1+driver","driver_api_version":"5.0.0","driver_schema_version":"2.0.0","engine_version":"26","falco_version":"0.36.1","libs_version":"0.13.2","plugin_api_version":"3.1.0"}
```

- System info:

```
{
  "machine": "x86_64",
  "nodename": "some node",
  "release": "5.15.107+",
  "sysname": "Linux",
  "version": "#1 SMP Sat May 20 09:38:40 UTC 2023"
}
```
- Cloud provider or hardware configuration:
- OS:
```
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
- Kernel:
```
Linux somenode 5.15.107+ #1 SMP Sat May 20 09:38:40 UTC 2023 x86_64 GNU/Linux
```
- Installation method:

- Custom K8S manifest using standard container image
- For debugging built falco from source as DEBUG release (with ASSERTIONS stripped)

**Additional context**

**Debug run output**
```
Wed Oct 18 01:15:43 2023: Falco version: 0.36.1 (x86_64)
Wed Oct 18 01:15:43 2023: Falco initialized with configuration file: /tmp/make/falco/falco.yaml
Wed Oct 18 01:15:43 2023: Loading rules from file /etc/falco/falco_rules.yaml
Wed Oct 18 01:15:43 2023: Loading rules from file /etc/falco/falco_rules.local.yaml
Wed Oct 18 01:15:43 2023: Loading rules from file /etc/falco/rules.d/custom.yaml
Wed Oct 18 01:15:43 2023: The chosen syscall buffer dimension is: 8388608 bytes (8 MBs)
Wed Oct 18 01:15:43 2023: Starting health webserver with threadiness 2, listening on port 8765
Wed Oct 18 01:15:43 2023: Loaded event sources: syscall
Wed Oct 18 01:15:43 2023: Enabled event sources: syscall
Wed Oct 18 01:15:43 2023: Opening 'syscall' source with BPF probe. BPF probe path: /root/.falco/falco-bpf.o
```

**Stack trace**
Is too long so [pastebin here](https://pastebin.com/wXMhaZZw)

**Comment by max-frank** [comment]
@FedeDP I've replaced https://github.com/falcosecurity/libs/blob/5b4e28d44f9c87aa23f24c53c6bfb696b2b47635/userspace/libsinsp/sinsp.cpp#L1436-L1447 with the suggested lines and compiled the debug build.

I've got it running now now I can just wait and see if it will crash or not.

**Comment by FedeDP** [comment]
@max-frank you are the best tester out there! Thank you very much! Fingers :crossed_fingers:

**Comment by FedeDP** [comment]
@max-frank any new crash after applying the patch? :)

**Comment by max-frank** [comment]
@FedeDP Ok so its been officially around 24h and can report not crash yet
<img width="1184" alt="image" src="https://github.com/falcosecurity/falco/assets/1375575/7d38e84d-c8e6-4d88-8ba7-e4a95ce3d716">

**Comment by FedeDP** [comment]
/milestone 0.36.2

We are arranging a 0.36.2 patch release for Falco to fix this :) 
See https://github.com/falcosecurity/falco/issues/2885

**Comment by max-frank** [comment]
Very nice much appreciated

**Comment by max-frank** [comment]
Just deployed it in our local deployment test cluster.  No crashes for syscall falco yet with the release candidate.

We are still experiencing issues with gvisor crashing on startup on our GKE 1.26 configuration, but that is a pre-existing issue I did not have time to debug yet. So I don't know if its a configuration issue on our end or an issue in falco.

**Comment by FedeDP** [comment]
Thanks for your feedback, it is really valuable! And thanks for helping us spot this issue!
About gvisor, yes please let's track it in a new issue eventually!
Thank you very much!

**Comment by FedeDP** [comment]
This is fixed in Falco 0.36.2, just released!
Thank you very much!
/close
